rest_api/controller/search.py

```diff
@@ -1,216 +1,74 @@
-            status_code=404, detail=f"Could not get Finder with ID {model_id}. Available IDs: {list(FINDERS.keys())}"
-        )
-
-    results = []
-    for question in request.questions:
-        if request.filters:
-            # put filter values into a list and remove filters with null value
-            filters = {}
-            for key, values in request.filters.items():
-                if values is None:
-                    continue
-                if not isinstance(values, list):
-                    values = [values]
-                filters[key] = values
-            logger.info(f" [{datetime.now()}] Request: {request}")
-        else:
-            filters = {}
-
-        result = finder.get_answers_via_similar_questions(
-            question=question, top_k_retriever=request.top_k_retriever, filters=filters,
-        )
-        results.append(result)
-
-    elasticapm.set_custom_context({"results": results})
-    logger.info(json.dumps({"request": request.dict(), "results": results}))
-
-    return {"results": results}
-
-
-@router.post("/models/{model_id}/query", response_model=Dict[str, Any], response_model_exclude_unset=True)
-def query(model_id: int, query_request: Dict[str, Any], top_k_reader: int = DEFAULT_TOP_K_READER):
-    with doc_qa_limiter.run():
-        start_time = time.time()
-        finder = FINDERS.get(model_id, None)
-        if not finder:
-            raise HTTPException(
-                status_code=404, detail=f"Could not get Finder with ID {model_id}. Available IDs: {list(FINDERS.keys())}"
-            )
-
-        question_request = Question.from_elastic_query_dsl(query_request, top_k_reader)
-
-        answers = search_documents(finder, question_request, start_time)
-        response: Dict[str, Any] = {}
-        if answers and len(answers) > 0:
-            response = AnswersToIndividualQuestion.to_elastic_response_dsl(dict(answers[0]))
-
-        return response
-
-
-def search_documents(finder, question_request, start_time) -> List[AnswersToIndividualQuestion]:
-    results = []
-    for question in question_request.questions:
-        if question_request.filters:
-            # put filter values into a list and remove filters with null value
-            filters = {}
-            for key, values in question_request.filters.items():
-                if values is None:
-                    continue
-                if not isinstance(values, list):
-                    values = [values]
-                filters[key] = values
-            logger.info(f" [{datetime.now()}] Request: {question_request}")
-        else:
-            filters = {}
-
-        result = finder.get_answers(
-            question=question,
-            top_k_retriever=question_request.top_k_retriever,
-            top_k_reader=question_request.top_k_reader,
-            filters=filters,
-        )
-        results.append(result)
-    elasticapm.set_custom_context({"results": results})
+
+class Request(BaseModel):
```

Comment:
We can add `schema_extra` to add sample example for request and response on swagger
https://fastapi.tiangolo.com/tutorial/schema-extra-example/